Aida-vm crashes for testnet block 9159681 when using `--validate-tx`

When running
```
go run ./cmd/aida-vm --aida-db /var/opera/herbert/testnet/aida-db --vm-impl evmzero --validate-tx 9159681 9159681
```
Aida crashes with the following error:
```
2023/12/01 10:14:49 NOTICE: Total elapsed time: 0s; last block 0; total transaction rate ~0.00 Tx/s, ~0.00 MGas/s
panic: record-replay: error getting code 0x2d54e0ca43c59da854b37f956119f7d312d47902e03b4c579ec06d4b74fd7ad0: leveldb: closed

goroutine 215 [running]:
github.com/Fantom-foundation/Substate.(*SubstateDB).GetCode(0xc007224040, {0x2d, 0x54, 0xe0, 0xca, 0x43, 0xc5, 0x9d, 0xa8, 0x54, ...})
        /home/herbert/go/pkg/mod/github.com/!fantom-foundation/!substate@v0.0.0-20231129093658-a074f38004d0/substate_db.go:124 +0x268
github.com/Fantom-foundation/Substate.(*SubstateAlloc).SetRLP(0xc006e1fa70, {{0xc0072442d0, 0x3, 0x4}, {0xc000147d60, 0x3, 0x4}}, 0x30?)
        /home/herbert/go/pkg/mod/github.com/!fantom-foundation/!substate@v0.0.0-20231129093658-a074f38004d0/substate_rlp.go:82 +0x177
github.com/Fantom-foundation/Substate.(*Substate).SetRLP(0xc006e1fa70, 0xc000040600, 0x99b?)
        /home/herbert/go/pkg/mod/github.com/!fantom-foundation/!substate@v0.0.0-20231129093658-a074f38004d0/substate_rlp.go:404 +0x1b5
github.com/Fantom-foundation/Substate.parseTransaction(0x0?, {{0xc006a5e0f0, 0x12, 0x12}, {0xc000356000, 0x99b, 0x99b}})
        /home/herbert/go/pkg/mod/github.com/!fantom-foundation/!substate@v0.0.0-20231129093658-a074f38004d0/substate_db.go:328 +0x1ab
github.com/Fantom-foundation/Substate.NewSubstateIterator.func2()
        /home/herbert/go/pkg/mod/github.com/!fantom-foundation/!substate@v0.0.0-20231129093658-a074f38004d0/substate_db.go:523 +0x145
created by github.com/Fantom-foundation/Substate.NewSubstateIterator in goroutine 210
        /home/herbert/go/pkg/mod/github.com/!fantom-foundation/!substate@v0.0.0-20231129093658-a074f38004d0/substate_db.go:520 +0x28c
panic: record-replay: error getting code 0x76276ac0f1f95f82f789e8cfe9efea75e9b30d39dde75cfab9e27d441b6fe538: leveldb: closed

goroutine 214 [running]:
github.com/Fantom-foundation/Substate.(*SubstateDB).GetCode(0xc007224040, {0x76, 0x27, 0x6a, 0xc0, 0xf1, 0xf9, 0x5f, 0x82, 0xf7, ...})
        /home/herbert/go/pkg/mod/github.com/!fantom-foundation/!substate@v0.0.0-20231129093658-a074f38004d0/substate_db.go:124 +0x268
github.com/Fantom-foundation/Substate.(*SubstateAlloc).SetRLP(0xc007211658, {{0xc00558c2d0, 0x3, 0x4}, {0xc007720800, 0x3, 0x4}}, 0x30?)
        /home/herbert/go/pkg/mod/github.com/!fantom-foundation/!substate@v0.0.0-20231129093658-a074f38004d0/substate_rlp.go:82 +0x177
github.com/Fantom-foundation/Substate.(*Substate).SetRLP(0xc007211650, 0xc0076ce280, 0xf33?)
        /home/herbert/go/pkg/mod/github.com/!fantom-foundation/!substate@v0.0.0-20231129093658-a074f38004d0/substate_rlp.go:405 +0x205
github.com/Fantom-foundation/Substate.parseTransaction(0x0?, {{0xc006a5e288, 0x12, 0x12}, {0xc0004c2000, 0xf33, 0xf33}})
        /home/herbert/go/pkg/mod/github.com/!fantom-foundation/!substate@v0.0.0-20231129093658-a074f38004d0/substate_db.go:328 +0x1ab
github.com/Fantom-foundation/Substate.NewSubstateIterator.func2()
        /home/herbert/go/pkg/mod/github.com/!fantom-foundation/!substate@v0.0.0-20231129093658-a074f38004d0/substate_db.go:523 +0x145
created by github.com/Fantom-foundation/Substate.NewSubstateIterator in goroutine 210
        /home/herbert/go/pkg/mod/github.com/!fantom-foundation/!substate@v0.0.0-20231129093658-a074f38004d0/substate_db.go:520 +0x28c
exit status 2
```
Without the `--validate-tx` flag the command runs successfully.

This issue was discovered while investigating #838.

--- a/executor/executor.go
+++ b/executor/executor.go
@@ -427,11 +427,16 @@ func (e *executor[T]) runParallelTransaction(params Params, processor Processor[
 	// An event for signaling an abort of the execution.
 	abort := utils.MakeEvent()
 
+	var wg sync.WaitGroup
 	// Start one go-routine forwarding transactions from the provider to a local channel.
 	var forwardErr error
 	transactions := make(chan *TransactionInfo[T], 10*numWorkers)
+	wg.Add(1)
 	go func() {
-		defer close(transactions)
+		defer func() {
+			close(transactions)
+			wg.Done()
+		}()
 		abortErr := errors.New("aborted")
 		err := e.provider.Run(params.From, params.To, func(tx TransactionInfo[T]) error {
 			select {
@@ -448,7 +453,7 @@ func (e *executor[T]) runParallelTransaction(params Params, processor Processor[
 
 	// Start numWorkers go-routines processing transactions in parallel.
 	var cachedPanic atomic.Value
-	var wg sync.WaitGroup
+
 	wg.Add(numWorkers)
 	workerErrs := make([]error, numWorkers)
 	e.log.Debugf("Starting %v workers run on Transaction granularity...", numWorkers)
@@ -460,8 +465,8 @@ func (e *executor[T]) runParallelTransaction(params Params, processor Processor[
 					abort.Signal() // stop forwarder and other workers too
 					cachedPanic.Store(r)
 				}
+				wg.Done()
 			}()
-			defer wg.Done()
 			for {
 				select {
 				case tx := <-transactions:
@@ -483,6 +488,7 @@ func (e *executor[T]) runParallelTransaction(params Params, processor Processor[
 			}
 		}(i)
 	}
+
 	wg.Wait()
 
 	if r := cachedPanic.Load(); r != nil {

